File: cpp/76_counting_summations.cpp
#include <iostream>

int counting_summations(){
    int summations[101] = {0};
    summations[0] = 1;
    // summing with integers i [1, 99]
    for (int i = 1; i < 100; i++){
        // summing to i integers j [1, 100]
        for (int j = i; j <= 100; j++){
            // count sums to j using i as largest integer in sum
            summations[j] += summations[j - i];
        }
    }
    return summations[100];
}

int main(){
    std::cout << counting_summations() << "\n";
}

File: c#/_76_CountingSummations.cs
using System;

public class _76_CountingSummations{
	public static int Euler76(){
		int[] sums = new int[101];
		sums[0] = 1;
        // summing positive integers less than 100 (exclusive)
        // i is current largest positive integer used
		for (int i = 1; i < 100; i++){
            // summing to positive integers up to 100 (inclusive)
			for (int j = i; j <= 100; j++){
				sums[j] += sums[j - i];
			}
		}
		return sums[100];
	}
	
	public static void Main(){
		Console.WriteLine(Euler76());
	}
}
